opensky.ipynb

```diff
@@ -310,98 +289,27 @@
-    "content = json.loads(req.content)\n",
-    "states = content['states']\n",
-    "df = pd.DataFrame(states, columns=cols)\n",
-    "df['timestamp'] = content['time']\n",
-    "df.to_sql('opensky', conn, index=False, if_exists='append')\n",
-    "```"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "The resulting `opensky.db` file was then transformed into Web Mercator coordinates, split per flight, and exported to Parquet format, using the code below.  This process took about 7 minutes on a MacBook Pro laptop."
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "```python\n",
-    "import sqlite3, pandas as pd, numpy as np, holoviews as hv, datashader.utils as du\n",
-    "\n",
-    "def transform_coords(df):\n",
-    "    df=df.copy()\n",
-    "    df.loc[:, 'longitude'], df.loc[:, 'latitude'] = \\\n",
-    "        du.lnglat_to_meters(df.longitude,df.latitude)\n",
-    "    return df\n",
-    "\n",
-    "def split_flights(df):\n",
-    "    df = df.copy().reset_index(drop=True)\n",
-    "    df = df[np.logical_not(df.time_position.isnull())]\n",
-    "    empty=df[:1].copy()\n",
-    "    empty.loc[0, :] = 0\n",
-    "    empty.loc[0, 'origin'] = ''\n",
-    "    empty.loc[0, 'latitude'] = np.NaN\n",
-    "    empty.loc[0, 'longitude'] = np.NaN\n",
-    "    paths = []\n",
-    "    for gid, group in df.groupby('icao24'):\n",
-    "        times = group.time_position\n",
-    "        splits = np.split(group.reset_index(drop=True), np.where(times.diff()>600)[0])\n",
-    "        for split_df in splits:\n",
-    "            if len(split_df) > 20:\n",
-    "                paths += [split_df, empty]\n",
-    "    split = pd.concat(paths,ignore_index=True)\n",
-    "    split['ascending'] = split.vertical_rate>0\n",
-    "    return split\n",
-    "\n",
-    "# Load the data from a SQLite database and project into Web Mercator (1.5 min)\n",
-    "DB='../data/opensky.db'\n",
-    "conn = sqlite3.connect(DB)\n",
-    "df = transform_coords(pd.read_sql(\"SELECT * from flights\", conn))\n",
-    "\n",
-    "# Split into groups by flight (6 min)\n",
-    "flightpaths = split_flights(df)\n",
-    "\n",
-    "# Remove unused columns and declare categoricals\n",
-    "flightpaths = flightpaths[['longitude', 'latitude', 'origin', 'ascending', 'velocity']]\n",
-    "flightpaths['origin']    = flightpaths.origin.astype('category')\n",
-    "flightpaths['ascending'] = flightpaths.ascending.astype('bool')\n",
-    "\n",
-    "# Export to Parquet\n",
-    "args = dict(engine=\"fastparquet\", compression=\"snappy\", has_nulls=False, write_index=False)\n",
-    "flightpaths.to_parquet(\"../data/opensky.parq\", **args)\n",
-    "```"
+    "This data was obtained by running a cron job with the following script running at one-minute intervals over a four-day period using the script [on the wiki](https://github.com/pyviz/datashader/wiki/OpenSky-download-script)\n"
```

Comment:
Presumably the script shouldn't be on the wiki, but in the project dir, but it looks like this .ipynb doesn't have a directory yet?  If so the script should be moved into the directory once it exists.

Reply:
Agreed.